libs/cart/src/models/cart-item.ts

```diff
@@ -15,8 +15,21 @@ export interface DaffCartItem {
   sku: string;
   name: string;
   qty: number;
+  /**
+   * The price per item, not including cart discounts.
+   */
   price: number;
+  /**
+   * The total for the entire row, not including cart discounts.
+   */
   row_total: number;
+  /**
+   * The discount per item.
+   */
```

Comment:
After a lengthy discussion, this is definitely not possible in the manner examined in this PR. There are some nasty edge-cases with rounding and the actually expected behavior for rounding isn't consistent for every platform,  making this field either nullable or not per platform which is not a good enough interface for us to describe in Daffodil.

For example, consider the scenario:

```txt
round(A * B * C) ?= round(A*B) * C
```


One can formally prove that these are not necessarily the same, an exercise left for the reader.

Magento falls into the `round(A * B * C)` which means that if we attempt to compute a discount per item, we'll be left with half-pennies in various scenarios which is both numerically unsafe and potentially confusing.

Reply:
It seems like the only perfectly accurate option is to expose every single price-related field on the backend and directly map to Daffodil fields.

Reply:
I think we can just never do division with discount fields

Reply:
Generally speaking, I'd argue doing any division on the client is going to cause us headaches unless we're extremely careful.